main.cc

```diff
@@ -146,6 +146,62 @@ read_config(bpo::variables_map& opts, db::config& cfg) {
         return make_exception_future<>(ep);
     });
 }
+
+class sighup_handler {
```

Comment:
How about a short blurb explaining how the user might use this class?  When to call which public function, etc.

Reply:
added (in a new commit, to be squahed later)